maestro-ios-xctest-runner/maestro-driver-iosUITests/Routes/Handlers/ViewHierarchyHandler.swift

```diff
@@ -0,0 +1,182 @@
+import FlyingFox
+import XCTest
+import os
+
+extension NSException: Error {}
+
+@MainActor
+struct ViewHierarchyHandler: HTTPHandler {
+
+    let maxDepth = 60
+    let springboardApplication = XCUIApplication(bundleIdentifier: "com.apple.springboard")
+
+    private let logger = Logger(
+        subsystem: Bundle.main.bundleIdentifier!,
+        category: String(describing: Self.self)
+    )
+
+    func handleRequest(_ request: FlyingFox.HTTPRequest) async throws -> HTTPResponse {
+        guard let requestBody = try? JSONDecoder().decode(ViewHierarchyRequest.self, from: request.body) else {
+            let errorInfo = [
+                "errorMessage": "incorrect request body provided",
+            ]
+            let body = try? JSONEncoder().encode(errorInfo)
+            return HTTPResponse(statusCode: .badRequest, body: body ?? Data())
+        }
+
+        let appId = requestBody.appId
+
+        do {
+            let viewHierarchy = try logger.measure(message: "View hierarchy snapshot for \(appId)") {
+                try getViewHierarchy(appId: appId)
+            }
+
+            let body = try JSONEncoder().encode(viewHierarchy)
+
+            // TODO: Remove debug print here
+            print(String(data: body, encoding: .utf8)!)
```

Comment:
should we remove or PR still WIP?

Reply:
Yes I have shifted to draft. I'll be adding more improvements and addressing comments in upcoming PR